### DISCUSSION TITLE
Use natural sort order for album titles

### DISCUSSION THREAD

**Discussion by MelHarbour** [music-assistant/support#2499] [Feature requests and ideas]
Some albums (compilations particularly) have a lot of entries in their series, and a natural sort order would be more intuitive:

https://en.wikipedia.org/wiki/Natural_sort_order

So, for example:

Now That's What I Call Music 1
Now That's What I Call Music 2
Now That's What I Call Music 11

should really be in that order, but in reality 11 will display before 2.

Probably the cleanest way of doing this is adding a new collation to the SQLite database, such as explained here: https://sqlite.org/forum/forumpost/cff93834d2b07da5 but I think you may not be able to modify the collation of a SQLite table after creation.

**Comment by marcelveldt** [comment]
We already should have this for the library listings. Where are you seeing the wrong sort order?

**Reply by MelHarbour** [reply]
Artists -> Various Artists ->

![image](https://github.com/music-assistant/hass-music-assistant/assets/4092769/2433d545-ed93-4ba8-b6bc-99e38571bf5f)

**Reply by OzGav** [reply]
Have you tried the sort name option?

**Reply by MelHarbour** [reply]
Yes, same behaviour. And, to be fair, looking at the code, you'd probably expect that. If the collation on the table worked for one, it would work for both.

**Reply by marcelveldt** [reply]
That data is not from the DB (well, partially it is) and thus not sorted the db but by the frontend
What did you select in the sort options ?

**Reply by MelHarbour** [reply]
I've tried Alphabetically and Sort Name. It's in Various Artists and I've got something like 153 albums in there. The data looks like it's being fetched paged, which would suggest server side sorting.

**Reply by marcelveldt** [reply]
Nah, the artist tracks and album tracks are not server side paged and/or sorted. That is all frontend.
We just need to adjust the sort function in the frontend here to account for natural sort: 
https://github.com/music-assistant/frontend/blob/main/src/components/ItemsListing.vue#L973

**Comment by MelHarbour** [comment]
Closing as implemented
